pkg/kubernetes/pod/v1alpha1/pod.go

```diff
@@ -142,3 +142,35 @@ func IsNil() Predicate {
 func (p *Pod) GetAPIObject() *corev1.Pod {
 	return p.object
 }
+
+// FromList created a PodList with provided api podlist
+func FromList(pods *corev1.PodList) *PodList {
+	pl := ListBuilderForAPIList(pods).List()
+	return pl
+}
+
+// GetNodeNames returns the nodes on which pods are running
+func (pl *PodList) GetNodeNames() []string {
+	nodeNames := []string{}
+	for _, p := range pl.items {
+		nodeNames = append(
+			nodeNames,
+			p.object.Spec.NodeName,
+		)
+	}
+	return nodeNames
+}
+
+// IsMatchesAnyNode checks the PodList is running on the provided nodes
+func (pl *PodList) IsMatchesAnyNode(nodes []string) bool {
+	var result bool
+	for _, p := range pl.items {
+		result = false
+		for _, node := range nodes {
+			if p.object.Spec.NodeName == node {
+				result = true
+			}
+		}
+	}
```

Comment:
Can we make use of Predicate ?
```go
func IsMatchNode(node string) Predicate {
  return func(p *pod) bool {
    return p.object.Spec.NodeName == node
  }
}
```

Reply:
I want to use a Predicate but instead of comparing strings we have to compare slices of string and golang does not have any function for that.

Reply:
This predicate or bool function will be used from within the loop of `IsMatchNodeAny`

Reply:
@shubham14bajpai perhaps `reflect.DeepEqual()` do the job.